unittest/factory/residual.hpp

```diff
@@ -10,7 +10,6 @@
 #define CROCODDYL_RESIDUAL_FACTORY_HPP_
 
 #include "state.hpp"
-#include "residual.hpp"
```

Comment:
This also seems to be removed by mistake. The header includes our factory for creating different residual objects.

Reply:
Isn't that the same file it's being included in?

Reply:
Yes, you're right! Thanks for cleaning up!